app/src/main/kotlin/com/mapbox/vision/teaser/ar/ArMapActivity.kt

```diff
@@ -155,69 +145,65 @@ class ArMapActivity : AppCompatActivity(), MapboxMap.OnMapClickListener, OnMapRe
 
     override fun onMapReady(mapboxMap: MapboxMap) {
         this.mapboxMap = mapboxMap
-        mapboxMap.setStyle(Style.Builder().fromUri(MAP_STYLE)) {
+        mapboxMap.setStyle(Style.Builder().fromUri(Style.DARK)) {
             enableLocationComponent()
         }
 
         mapboxMap.addOnMapClickListener(this)
     }
 
     private fun getRoute(origin: Point, destination: Point) {
-        NavigationRoute.builder(this)
-            .accessToken(Mapbox.getAccessToken()!!)
-            .origin(origin)
-            .destination(destination)
-            .build()
-            .getRoute(object : Callback<DirectionsResponse> {
-                override fun onResponse(call: Call<DirectionsResponse>, response: Response<DirectionsResponse>) {
-                    if (response.body() == null || response.body()!!.routes().size == 0) {
-                        return
-                    }
-
-                    currentRoute = response.body()!!.routes()[0]
+        mapboxNavigation?.requestRoutes(
+            routeOptions = RouteOptions.builder()
+                .applyDefaultParams()
+                .accessToken(Mapbox.getAccessToken()!!)
+                .coordinates(listOf(origin, destination))
+                .build(),
+            routesRequestCallback = object : RoutesRequestCallback {
+                override fun onRoutesReady(routes: List<DirectionsRoute>) {
+                    currentRoute = routes.first()
 
                     // Draw the route on the map
                     if (navigationMapRoute == null) {
-                        navigationMapRoute = NavigationMapRoute(null, mapView, mapboxMap, R.style.NavigationMapRoute)
+                        navigationMapRoute = NavigationMapRoute.Builder(
+                            mapView,
+                            mapboxMap,
+                            this@ArMapActivity,
+                        )
+                            .withStyle(R.style.MapboxStyleNavigationMapRoute)
+                            .build()
                     } else {
                         navigationMapRoute?.updateRouteVisibilityTo(false)
                     }
                     navigationMapRoute?.addRoute(currentRoute)
                 }
 
-                override fun onFailure(call: Call<DirectionsResponse>, throwable: Throwable) {}
-            })
+                override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+                    TODO("not implemented")
+                }
+
+                override fun onRoutesRequestFailure(throwable: Throwable, routeOptions: RouteOptions) {
+                    TODO("not implemented")
```

Comment:
afair will cause a runtime crash if this code is hit, better just leave a comment here